### PR TITLE
Copy CA certificates to deployment containers

### PR DESCRIPTION
**Pull request by tsmetana**
This fixes a problem with AWS cloud provider complaining about missing root certificates and resulting failed API calls.

cc: @rootfs, @jsafrane

### CODE REVIEW

**Comment by jsafrane** [comment]
We should not copy certificates from the machine where we build images - it can be some random bot out of our control with keys at weird places. IMO you should pick the certificates you think are needed and push them to git.

File: snapshot/Makefile [hunk 1]
```diff
@@ -46,6 +50,15 @@ container: controller provisioner container-quick
 container-quick:
 	cp _output/bin/snapshot-controller deploy/docker/controller
 	cp _output/bin/snapshot-provisioner deploy/docker/provisioner
+	# Copy the root CA certificates -- cloudproviders need them
+	cp --parent -r /etc/pki/ca-trust deploy/docker/controller
```
Comment (rootfs) [inline snapshot/Makefile]: this does not always work especially if ca-certificates pkg is not locally installed

Comment (rootfs) [reply snapshot/Makefile]: this Dockerfile seems to work for me locally
```
FROM  busybox:glibc
CMD rpm -i https://www.rpmfind.net/linux/fedora/linux/development/rawhide/Everything/x86_64/os/Packages/c/ca-certificates-2017.2.20-6.fc28.noarch.rpm
CMD EXTRACT_UNSAFE_SYMLINKS=1 rpm -i ca-certificates-2017.2.20-6.fc28.noarch.rpm
```

File: snapshot/deploy/docker/controller/Dockerfile [hunk 2]
```diff
@@ -14,4 +14,9 @@
 
 FROM busybox:glibc
 COPY snapshot-controller /bin/snapshot-controller
+# Root CA certificates
+COPY /etc/pki/ca-trust /etc/pki/ca-trust
```
Comment (rootfs) [inline snapshot/deploy/docker/controller/Dockerfile]: this Dockerfile seems to work for me locally
```
FROM  busybox:glibc
CMD rpm -i https://www.rpmfind.net/linux/fedora/linux/development/rawhide/Everything/x86_64/os/Packages/c/ca-certificates-2017.2.20-6.fc28.noarch.rpm
CMD EXTRACT_UNSAFE_SYMLINKS=1 rpm -i ca-certificates-2017.2.20-6.fc28.noarch.rpm
```

Comment (tsmetana) [reply snapshot/deploy/docker/controller/Dockerfile]: I'm not sure I want to rely on rpmfind.net to get the package from: I don't know what would happen to the link once the package is obsoleted. Also -- if I have rpm I can be reasonably sure I have the CA certificates on the system too.

Comment (rootfs) [reply snapshot/deploy/docker/controller/Dockerfile]: @tsmetana updated with fedora pkg.

**Comment by tsmetana** [comment]
OK. I have capitulated: Adding the CA certificates to the repository and using the bundled copies to build the images. PTAL.

File: snapshot/deploy/docker/controller/Dockerfile (hunk 2 above)
Comment (rootfs) [reply snapshot/deploy/docker/controller/Dockerfile]: need to update /etc/ :)

Comment (tsmetana) [reply snapshot/deploy/docker/controller/Dockerfile]: Eh... How did I... nm. Will fix this, thanks.

**Comment by tsmetana** [comment]
Updated Dockerfiles to remove at least one layer. The CA files are stored in-tree, so no assumptions about the build system. Once merged I will push new Quay images, which should hopefully fix issue #558.

**Comment by rootfs** [comment]
/lgtm

cc @childsb for merge

**Pull request merged**